### PR TITLE
#27 make TxMessage data members public

### PR DESCRIPTION
**Pull request by HusseinAbdelhamid**
see #27

### CODE REVIEW

**Review by marius-meissner — CHANGES_REQUESTED** [review]
I don't think it's worth giving up OOP or stability and security principles just because of tests.
What about just adding getter methods, which return immutable borrows?
This would also have the advantage that these methods may also be helpful outside of tests.

**Comment by HusseinAbdelhamid** [comment]
> I don't think it's worth giving up OOP or stability and security principles just because of tests. What about just adding getter methods, which return immutable borrows? This would also have the advantage that these methods may also be helpful outside of tests.

Yes probably a better idea, so as an example to get the buff member we could implement a fn like this: 
```rust
  pub fn get_payload(&self) -> &Bytes {
        &self.buff
    }
```

File: src/tests/message.rs [hunk 1]
```diff
@@ -111,3 +111,15 @@ fn test_payload_greater_than_generic_type_args() {
     assert_eq!(message_2_0.unwrap_err(), MessageError::InvalidLength(5));
     assert_eq!(message_fd.unwrap_err(), MessageError::InvalidLength(23));
 }
+
+#[test]
```
Comment (marius-meissner) [inline src/tests/message.rs:115]: Even with unit test, nice :) :+1:

**Review by marius-meissner — APPROVED** [review]
LGTM

**Pull request merged**